pkg/volume/flexvolume/flexvolume.go

```diff
@@ -91,236 +94,291 @@ func (plugin *flexVolumePlugin) GetAccessModes() []api.PersistentVolumeAccessMod
-	fv := plugin.getVolumeSource(spec)
+func (plugin *flexVolumePlugin) WaitForAttach(spec *volume.Spec, timeout time.Duration) (string, error) {
+	call := NewDriverCallWithTimeout(plugin.getExecutable(), waitForAttachCmd, exec.New(), timeout)
+	call.AppendSpec(spec, plugin.host)
+
+	status, err := call.Run()
+	if err != nil {
+		return "", err
+	}
+	return status.Device, nil
+}
+
+func (plugin *flexVolumePlugin) GetDeviceMountPath(spec *volume.Spec) string {
+	call := NewDriverCall(plugin.getExecutable(), getDeviceMountPathCmd, exec.New())
+	call.AppendSpec(spec, plugin.host)
+
+	status, err := call.Run()
+	if isCmdNotSupportedErr(err) {
+		name := plugin.driverName
+		return plugin.host.GetPodVolumeDir(spec.PodUID, utilstrings.EscapeQualifiedNameForDisk(name), spec.Name())
+	} else if err != nil {
+		glog.Errorf("FlexVolume: driver bug in %s: returned an error on getdevicemountpath: %s", plugin.getExecutable(), err)
+		return ""
+	}
+	return status.Path
+}
+
+func (plugin *flexVolumePlugin) MountDevice(spec *volume.Spec, devicePath string, deviceMountPath string, mounter mount.Interface) error {
+	call := NewDriverCall(plugin.getExecutable(), mountDeviceCmd, exec.New())
+	call.AppendSpec(spec, plugin.host)
+	call.Append(devicePath)
+	call.Append(deviceMountPath)
+
+	_, err := call.Run()
+	if isCmdNotSupportedErr(err) {
+		return plugin.defaultMountDevice(spec, devicePath, deviceMountPath, mounter)
+	}
+	return err
+}
+
+func (plugin *flexVolumePlugin) defaultMountDevice(spec *volume.Spec, devicePath string, deviceMountPath string, mounter mount.Interface) error {
+	volSource := getVolumeSource(spec)
+
+	options := make([]string, 0)
+
+	if spec.ReadOnly {
+		options = append(options, "ro")
+	} else {
+		options = append(options, "rw")
+	}
+
+	os.MkdirAll(deviceMountPath, 0750)
+	// Mount not supported by driver. Use core mounting logic.
+	err := mounter.Mount(devicePath, deviceMountPath, volSource.FSType, options)
+	if err != nil {
+		glog.Errorf("Failed to mount the volume: %s, device: %s, error: %s", spec.Name(), devicePath, err.Error())
+		return err
+	}
+	return nil
+}
+
+// Detacher interface
+
+func (plugin *flexVolumePlugin) Detach(deviceName, hostName string) error {
+	call := NewDriverCall(plugin.getExecutable(), detachCmd, exec.New())
+	call.Append(deviceName)
+	call.Append(hostName)
+
+	_, err := call.Run()
+	return err
+}
+
+func (plugin *flexVolumePlugin) WaitForDetach(devicePath string, timeout time.Duration) error {
+	call := NewDriverCallWithTimeout(plugin.getExecutable(), waitForDetachCmd, exec.New(), timeout)
+	call.Append(devicePath)
+
+	_, err := call.Run()
+	return err
+}
+
+func (plugin *flexVolumePlugin) UnmountDevice(deviceMountPath string, mounter mount.Interface) error {
+	call := NewDriverCall(plugin.getExecutable(), unmountDeviceCmd, exec.New())
+	call.Append(deviceMountPath)
+
+	_, err := call.Run()
+	if isCmdNotSupportedErr(err) {
```

Comment:
Same here.. in case of error return, do not use default mount/unmount routines..